dipy/reconst/dti.py

```diff
@@ -824,12 +878,17 @@ def fit(self, data, mask=None):
             dti_params = params_in_mask.reshape(out_shape)
             if self.return_S0_hat:
                 S0_params = model_S0.reshape(out_shape[:-1])
+            if extra is not None:
+                self.extra = extra.reshape(data.shape)
```

Comment:
I would recommend to initialize `self.extra = {}` in the init function

Reply:
good idea.

The reason that the 'extra' field was a numpy array, is that at various times arrays become flattened due to masks and due to wrappers for iteration over voxels, which makes it quite tricky to generalize these extra returns to something other than a numpy array with the same shape as the data.

Thinking through possible use cases, it is hard to see where an extra return wouldn't be the same shape as the data array, so for now I will try to generalize to a dictionary of items that are the same shape as the data array. This allows to return whether a data-point is robust, to return the leverages of linear problems, the weights used in various fitting functions, the fitting residuals (although I see no point in returning these).